datasafe-storage/datasafe-storage-impl-fs/src/test/java/de/adorsys/datasafe/business/impl/storage/FileSystemStorageServiceTest.java

```diff
@@ -37,6 +46,75 @@ void prepare(@TempDir Path dir) {
         );
     }
 
+    @Test
+    void objectExists() {
+        createFileWithMessage();
+        assertThat(storageService.objectExists(storageService.list(root).findFirst().get()));
+    }
+
+    @Test
+    void listEmpty() {
+        Path nonExistingFile = storageDir.resolve(UUID.randomUUID().toString());
+        AbsoluteLocation<PrivateResource> nonExistingFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(nonExistingFile.toUri()));
+        assertThat(storageService.list(nonExistingFileLocation).collect(Collectors.toList()).isEmpty());
+    }
+
+    @SneakyThrows
+    @Test
+    void resolveWithMkDirs() {
+        Path nonExistingFolder = storageDir.resolve(UUID.randomUUID().toString());
+        Path newFile = nonExistingFolder.resolve(UUID.randomUUID().toString());
+        AbsoluteLocation<PrivateResource> newFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(newFile.toUri()));
+
+        try (OutputStream os = storageService.write(newFileLocation)) {
+            os.write(MESSAGE.getBytes());
+        }
+        assertThat(storageService.objectExists(newFileLocation));
+    }
+
+    @SneakyThrows
+    @Test
+    void writeWithException() {
+        Path realRoot = Paths.get("/");
+        Path beforeRoot = realRoot.resolve("..");
+        AbsoluteLocation<PrivateResource> newFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(beforeRoot.toUri()));
+
+        assertThatThrownBy(() -> {
+            try (OutputStream os = storageService.write(newFileLocation)) {
+                os.write(MESSAGE.getBytes());
+            }
+        });
+    }
+
+    @SneakyThrows
+    @Test
+    void readWithException() {
+        Path beforeRoot = Paths.get("..").resolve(UUID.randomUUID().toString());
+        AbsoluteLocation<PrivateResource> newFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(beforeRoot.toUri()));
+
+        assertThatThrownBy(() -> {
+            try (InputStream is = storageService.read(newFileLocation)) {
+                StringWriter writer = new StringWriter();
+                String encoding = StandardCharsets.UTF_8.name();
+                IOUtils.copy(is, writer, encoding);
+                log.warn("found: " + writer.toString());
+            }
+        });
+    }
+
+    @SneakyThrows
+    @Test
+    void listDotFilesToo() {
+        Path dotFile = storageDir.resolve(".dotfile");
+        AbsoluteLocation<PrivateResource> newFileLocation = new AbsoluteLocation<>(BasePrivateResource.forPrivate(dotFile.toUri()));
+
+        assertThat(storageService.list(root).collect(Collectors.toList()).isEmpty());
+        try (OutputStream os = storageService.write(newFileLocation)) {
+            os.write(MESSAGE.getBytes());
+        }
+        assertThat(!storageService.list(root).collect(Collectors.toList()).isEmpty());
```

Comment:
assertThat(storageService.list(root)).isNotEmpty()

Reply:
sorry for all the assertThat errors. Just got it wrong in the first time. Now I checked it.